File: 프로그래머스/2/42577. 전화번호 목록/전화번호 목록.java
import java.util.*;

class Solution {
    public boolean solution(String[] phone_book) {
        List<String> phones=Arrays.asList(phone_book);
        Collections.sort(phones);
        boolean flag=true;
        for(int i=0;i<phones.size()-1;i++){
            if(phones.get(i+1).indexOf(phones.get(i))==0){
                flag=false;
            }
        }
        if(flag==false){
            return false;
        }
        else{
            return true;
        }
    }
}

File: 프로그래머스/2/42577. 전화번호 목록/전화번호 목록.py
def solution(phone_book):
    phone_book.sort()
    flag=True
    for i in range(len(phone_book)-1):
        if phone_book[i+1].startswith(phone_book[i]):
            flag= False
            break
    if not flag: return False
    else: return True
    